near-plugins/tests/upgradable.rs

```diff
@@ -0,0 +1,681 @@
+
+    // Verify the staging timestamp was removed along with the staged code.
+    setup.assert_staging_timestamp(None).await;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_staged_code() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox().await?;
+    let owner = worker.dev_create_account().await?;
+    let setup = Setup::new(
+        worker,
+        Some(owner.id().clone()),
+        Some(sdk_duration_from_secs(42)),
+    )
+    .await?;
+
+    // No code staged.
+    let staged = setup
+        .upgradable_contract
+        .up_staged_code(&setup.unauth_account)
+        .await?;
+    assert_eq!(staged, None);
+
+    // Stage code.
+    let code = vec![1, 2, 3];
+    let res = setup
+        .upgradable_contract
+        .up_stage_code(&owner, code.clone())
+        .await?;
+    assert_success_with_unit_return(res);
+
+    // Some code is staged.
+    let staged = setup
+        .upgradable_contract
+        .up_staged_code(&setup.unauth_account)
+        .await?
+        .expect("Code should be staged");
+    assert_eq!(staged, code);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_staged_code_hash() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox().await?;
+    let owner = worker.dev_create_account().await?;
+    let setup = Setup::new(
+        worker,
+        Some(owner.id().clone()),
+        Some(sdk_duration_from_secs(42)),
+    )
+    .await?;
+
+    // No code staged.
+    let staged_hash = setup
+        .upgradable_contract
+        .up_staged_code_hash(&setup.unauth_account)
+        .await?;
+    assert_eq!(staged_hash, None);
+
+    // Stage code.
+    let code = vec![1, 2, 3];
+    let res = setup
+        .upgradable_contract
+        .up_stage_code(&owner, code.clone())
+        .await?;
+    assert_success_with_unit_return(res);
+
+    // Some code is staged.
+    let staged_hash = setup
+        .upgradable_contract
+        .up_staged_code_hash(&setup.unauth_account)
+        .await?
+        .expect("Code should be staged");
+    let code_hash = convert_code_to_crypto_hash(code.as_slice());
+    assert_eq!(staged_hash, code_hash);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_deploy_code_without_delay() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox().await?;
+    let owner = worker.dev_create_account().await?;
+    let setup = Setup::new(worker.clone(), Some(owner.id().clone()), None).await?;
+
+    // Stage some code.
+    let code = vec![1, 2, 3];
+    let res = setup
+        .upgradable_contract
+        .up_stage_code(&owner, code.clone())
+        .await?;
+    assert_success_with_unit_return(res);
+    setup.assert_staged_code(Some(code)).await;
+
+    // Deploy staged code.
+    let res = setup.upgradable_contract.up_deploy_code(&owner).await?;
+    assert_success_with_unit_return(res);
+
+    Ok(())
+}
+
+/// TODO stage code that corresponds to a valid contract. Call a method of that contract to verify
+/// the staged code was deployed.
```

Comment:
Carrying over [this TODO](https://github.com/aurora-is-near/near-plugins/blob/0c25fab016096792112798d1ace1e2e5ff598b3a/near-plugins/src/upgradable.rs#L132). I suggest to tackle it in a separate PR.